src/runtime_src/core/pcie/tools/xbutil/xbutil.cpp

```diff
@@ -1186,37 +1186,70 @@ int xcldev::device::runTestCase(const std::string& py,
     std::string devInfoPath = name + "/test/";
     std::string xsaXclbinPath = getXsaPath(vendor) + devInfoPath;
     std::string dsaXclbinPath = dsaPath + devInfoPath;
-    std::string xrtTestCasePath = xrtPath + "test/" + py;
+    std::string xrtTestCasePath = xrtPath + "test/";
 
     output.clear();
 
     std::string xclbinPath;
     searchXsaAndDsa(m_idx, xsaXclbinPath, dsaXclbinPath, xclbinPath, output);
-    xclbinPath += xclbin;
-
-    if (stat(xrtTestCasePath.c_str(), &st) != 0 || stat(xclbinPath.c_str(), &st) != 0) {
-        //if bandwidth xclbin isn't present, skip the test
-        if(xclbin.compare("bandwidth.xclbin") == 0) {
-            output += "Bandwidth xclbin not available. Skipping validation.";
-            return -EOPNOTSUPP;
-        }
-        output += "ERROR: Failed to find ";
-        output += py;
-        output += " or ";
-        output += xclbin;
-        output += ", Shell package not installed properly.";
-        return -ENOENT;
-    }
+    std::string cmd;
+    
+    // if platform.json exists in the new shell pakcage then use the new testcase
+    // else fallback to old python test cases
+    
+    // NEW FLOW: runs if platform.json is available and the platform is not versal
+    // Note: we don't have cpp based versal bw testcase
+    auto json_exists = [xclbinPath]() 
+        {
+            return boost::filesystem::exists(xclbinPath + "platform.json") ? true : false;
+        };
+    if(json_exists() && xclbin.compare("versal_23_bandwidth.py") != 0) {
```

Comment:
We can have platform.json for versal platform?

Reply:
I don't think Heera's team has bw testcase for versal yet. We'll have to rely on python testcase until they add the versal bw

Reply:
do you mean that they can have verify kernel for versal, but no bandwidth for versal?

Reply:
Oh, I guess I'm mistaken. I don't think that verify kernel for versal works either. We should check and see if they are providing platform.json for versal. If not then  `xclbin.compare("versal_23_bandwidth.py") != 0` is redundant